### ISSUE TITLE
Why in training step, the loss is L1loss?

### ISSUE THREAD

**Issue by yanmengjia** [liruoteng/FlowNet#14]
I read the paper that the loss is EPE end point error. Can L1loss replace it?

**Comment by liruoteng** [comment]
Hi thank you for your questions.
If you look into the parameter of L1loss layer, it assigns l2_per_location to true, which means the layer computes L2 Loss rather than L1.

**Comment by yanmengjia** [comment]
So that means you use L1loss layer to compute the L2 loss of u and v respectively and sum them together to compute endpoint error?

**Comment by yanmengjia** [comment]
Sorry for reopening this issue. I have another question. I notice that you write your own eltwise layer and downsample layer. Can you explain a litter to me about those two layers?

**Comment by liruoteng** [comment]
Hi, 
1. In this case, it computes:  L2 loss = sqrt[(u_est - u_gt) + (v_est - v_gt)].
est indicates predicted flow, gt indicates ground-truth.

2. Which eltwise layer and downsample layer do you refer to ?

**Comment by yanmengjia** [comment]
Thank you for explaining L2 loss.
 I refer to eltwise layer and downsample layer you used in FlowNet/models/flownet/model_simple/train.prototxt.

**Comment by liruoteng** [comment]
I did not write my own eltwise layer and downsample layer in train.prototxt. That file is original actually.
